tests/integration/test_notebooks_gpu.py

```diff
@@ -175,31 +175,31 @@ def test_xdeepfm_integration(notebooks, syn_epochs, criteo_epochs, expected_valu
 @pytest.mark.integration
 @pytest.mark.gpu
 @pytest.mark.parametrize(
-    "size, epochs, expected_values, seed",
+    "size, steps, expected_values, seed",
     [
         (
             "1m",
-            10,
+            50000,
```

Comment:
how does this relate to epochs? will this be faster than what it was previously?

Reply:
@gramhagen Thanks for the review! Basically, `steps = num_train_samples * epochs / batch_size` and I'm using steps here just because the modern TF api accepts it. Benefit of using steps over epochs is you have finer control over how many batches to use for training. Useful when your training set is very big.

Previously, I converted epochs into steps manually from the notebook and passed it to model functions. Now I'm just accepting the steps directly.

Reply:
Btw, I'm having two tests failed. Are they known issue of the repo?

Reply:
mmm this is a weird error:
```
tests\unit\test_tf_utils.py ...E.E                                       [ 94%]
tests\unit\test_wide_deep_utils.py ..                                    [100%]

=================================== ERRORS ====================================
________________ ERROR at teardown of test_evaluation_log_hook ________________

tp = <class 'OSError'>, value = None, tb = None

    def reraise(tp, value, tb=None):
        try:
            if value is None:
                value = tp()
            if value.__traceback__ is not tb:
                raise value.with_traceback(tb)
>           raise value

C:\Anaconda\envs\reco_gpu\lib\site-packages\six.py:693: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
C:\Anaconda\envs\reco_gpu\lib\site-packages\six.py:693: in reraise
    raise value
C:\Anaconda\envs\reco_gpu\lib\site-packages\six.py:693: in reraise
    raise value
tests\conftest.py:25: in tmp
    yield td
C:\Anaconda\envs\reco_gpu\lib\tempfile.py:809: in __exit__
    self.cleanup()
C:\Anaconda\envs\reco_gpu\lib\tempfile.py:813: in cleanup
    _shutil.rmtree(self.name)
C:\Anaconda\envs\reco_gpu\lib\shutil.py:500: in rmtree
    return _rmtree_unsafe(path, onerror)
C:\Anaconda\envs\reco_gpu\lib\shutil.py:399: in _rmtree_unsafe
    onerror(os.rmdir, path, sys.exc_info())
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

path = 'C:\\Users\\recocat\\AppData\\Local\\Temp\\pytest-of-recocat\\pytest-9\\tmpccwa3tpo'
onerror = <function rmtree.<locals>.onerror at 0x00000221524BA2F0>

    def _rmtree_unsafe(path, onerror):
        try:
            if os.path.islink(path):
                # symlinks to directories are forbidden, see bug #1669
                raise OSError("Cannot call rmtree on a symbolic link")
        except OSError:
            onerror(os.path.islink, path, sys.exc_info())
            # can't continue even if onerror hook returns
            return
        names = []
        try:
            names = os.listdir(path)
        except OSError:
            onerror(os.listdir, path, sys.exc_info())
        for name in names:
            fullname = os.path.join(path, name)
            try:
                mode = os.lstat(fullname).st_mode
            except OSError:
                mode = 0
            if stat.S_ISDIR(mode):
                _rmtree_unsafe(fullname, onerror)
            else:
                try:
                    os.unlink(fullname)
                except OSError:
                    onerror(os.unlink, fullname, sys.exc_info())
        try:
>           os.rmdir(path)
E           OSError: [WinError 145] The directory is not empty: 'C:\\Users\\recocat\\AppData\\Local\\Temp\\pytest-of-recocat\\pytest-9\\tmpccwa3tpo'

```
looks like an error with shutil?

Reply:
Somehow only the Windows DSVM complains about TemporaryDirectory cleanup with pytest yield-fixture... while that works well on my local Windows machines (both laptop and workstation) and Linux DSVM.
Anyway I made a change for that and has been fixed now.